===================================================================
RCS file: /cvsroot/linux-libc-headers/include/linux/if.h,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- linux-libc-headers/include/linux/if.h	2004/05/01 16:22:33	1.3
+++ linux-libc-headers/include/linux/if.h	2004/05/06 21:58:24	1.4
@@ -88,6 +88,8 @@
  *	being very small might be worth keeping for clean configuration.
  */
 
+#ifndef _NET_IF_H
+
 struct ifmap 
 {
 	unsigned long mem_start;
@@ -99,6 +101,8 @@
 	/* 3 bytes spare */
 };
 
+#endif
+
 struct if_settings
 {
 	unsigned int type;	/* Type of physical device or protocol */
@@ -124,6 +128,8 @@
  * remainder may be interface specific.
  */
 
+#ifndef _NET_IF_H
+
 struct ifreq 
 {
 #define IFHWADDRLEN	6
@@ -149,6 +155,8 @@
 	} ifr_ifru;
 };
 
+#endif
+
 #define ifr_name	ifr_ifrn.ifrn_name	/* interface name 	*/
 #define ifr_hwaddr	ifr_ifru.ifru_hwaddr	/* MAC address 		*/
 #define	ifr_addr	ifr_ifru.ifru_addr	/* address		*/
@@ -174,6 +182,8 @@
  * must know all networks accessible).
  */
 
+#ifndef _NET_IF_H
+
 struct ifconf 
 {
 	int	ifc_len;			/* size of buffer	*/
@@ -183,6 +193,9 @@
 		struct	ifreq 		*ifcu_req;
 	} ifc_ifcu;
 };
+
+#endif
+
 #define	ifc_buf	ifc_ifcu.ifcu_buf		/* buffer address	*/
 #define	ifc_req	ifc_ifcu.ifcu_req		/* array of structures	*/
 
